modules/core/planar.ts

```diff
@@ -0,0 +1,97 @@
+import type { GeoProjection } from 'd3';
+import { geoMetersToLat } from '../geo';
+
+/**
+ * Parses a distance value from an OSM tag, which
+ * could either be a number, or a number with a unit
+ * (e.g. `15 m`).
+ * @returns value in metres or `undefined` if the value is invalid
+ */
+export function parseDistanceWithUnit(tagValue: string, defaultUnit: string) {
+  if (!tagValue) return undefined; // tag doesn't exist
+
+  // special case when feet and inches are used together.
+  const imperialCombo = tagValue.match(/([\d.]+) *('|′|ft|foot|feet) *([\d.]+) *("|″|in|inch|inches)/);
+  if (imperialCombo) {
+    const feet = +imperialCombo[1];
+    const inches = +imperialCombo[3];
+    return feet / 3.281 + inches / 39.37;
+  }
+
+  // the remaining code parses normal values (number + optional unit)
+
+  const unit = tagValue.match(/[^\d.]+/)?.[0].trim() || defaultUnit;
+  const value = parseFloat(tagValue);
+
+  if (Number.isNaN(value)) return undefined; // invalid value
```

Comment:
`parseFloat()` will also return infinity if `tagValue` is `Infinity`, so check `isFinite()` too.

Reply:
thanks, didn't realise that ! fixed now